gomint-server/src/main/java/io/gomint/server/command/internal/GamemodeCommand.java

```diff
@@ -0,0 +1,69 @@
+package io.gomint.server.command.internal;
+
+import io.gomint.ChatColor;
+import io.gomint.GoMint;
+import io.gomint.command.Command;
+import io.gomint.command.CommandOutput;
+import io.gomint.command.annotation.*;
+import io.gomint.command.validator.IntegerValidator;
+import io.gomint.command.validator.StringValidator;
+import io.gomint.command.validator.TargetValidator;
+import io.gomint.entity.EntityPlayer;
+import io.gomint.world.Gamemode;
+
+import java.util.Map;
+
+/**
+ * @author NycuRO
+ * @version 1.0
+ */
+@Name( "gamemode" )
+@Description( "Sets a player's game mode" )
+@Permission( "gomint.command.gamemode" )
+@Overload( {
+    @Parameter( name = "gameMode", validator = IntegerValidator.class),
+    @Parameter( name = "gameMode", validator = StringValidator.class, arguments = { "[a-zA-Z0-9ß\\-]+" })
+} )
+@Overload( {
+    @Parameter( name = "target", validator = TargetValidator.class),
+} )
+public class GamemodeCommand extends Command {
+
+    @Override
+    public CommandOutput execute( EntityPlayer player, String alias, Map<String, Object> arguments ) {
+        CommandOutput output = new CommandOutput();
+
+        Gamemode gamemode = (Gamemode) arguments.get( "gameMode");
+        EntityPlayer entityPlayer = (EntityPlayer) arguments.get( "target" );
+
+        // Spectator gamemode wasn't in showed commands, so i don't added it.
+        if (arguments.get( "gameMode" ) == "s") {
```

Comment:
Although you have an IntegerValidator, you just check if the gameMode argument is equal to a string and ignore possible integer inputs

Reply:
Strings should never be compared with `==` in java.